controller/tenant.go

```diff
@@ -246,7 +190,24 @@ func (c *TenantController) Show(ctx *app.ShowTenantContext) error {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 
-	return ctx.OK(&app.TenantSingle{Data: convertTenant(ctx, tenant, namespaces, c.clusterService.GetCluster)})
+	// check if any environment type is missing - should be provisioned
+	missing, _ := filterMissingAndExisting(namespaces)
```

Comment:
do you have a test for this `filterMissingAndExisting` func?

Reply:
only indirect tests - not direct
see: 
* when no namespace exists
`TestShowTenantOKWhenNoTenantExists`
`TestSetupTenantOKWhenNoTenantExists`

* when some namespaces exist 
`TestShowTenantWhenSomeNamespacesAreMissing`
`TestSetupTenantOKWhenAlreadyExists`

* when all namespace exist
`TestSetupConflictFailure`
`TestShowTenant`


in https://github.com/fabric8-services/fabric8-tenant/pull/742/files#diff-f9e0701950f6288288104f0f34160817